File: iCopySharp/Classes/PropertiesToXML.cs
using System;
using System.Xml;
using Microsoft.VisualBasic.CompilerServices;
using WIA;

namespace iCopy
{

    static class PropertiesToXMLType
    {

        public static void PropertiesToXML(WIA.Properties properties, XmlTextWriter writer)
        {
            foreach (Property prop in properties)
            {

                writer.WriteStartElement("Property");
                try
                {

                    writer.WriteAttributeString("ID", prop.PropertyID.ToString());
                    writer.WriteAttributeString("Name", prop.Name);
                    writer.WriteAttributeString("IsReadOnly", Conversions.ToString(prop.IsReadOnly));
                    writer.WriteAttributeString("Type", string.Format("{0}", (WiaPropertyType)prop.Type));
                    writer.WriteAttributeString("Value", prop.ToString());

                    writer.WriteStartElement("Subtype");
                    writer.WriteAttributeString("Type", string.Format("{0}", prop.SubType));
                    switch (prop.SubType)
                    {
                        case WiaSubType.UnspecifiedSubType:
                            {
                                writer.WriteAttributeString("Count", 0.ToString());
                                break;
                            }
                        case WiaSubType.FlagSubType:
                            {
                                writer.WriteAttributeString("Count", string.Format("{0}", prop.SubTypeValues.Count));
                                foreach (object o in prop.SubTypeValues)
                                {
                                    writer.WriteStartElement("Flag");
                                    writer.WriteString(o.ToString());
                                    writer.WriteEndElement();
                                }

                                break;
                            }
                        case WiaSubType.ListSubType:
                            {
                                writer.WriteAttributeString("Count", string.Format("{0}", prop.SubTypeValues.Count));
                                foreach (object o in prop.SubTypeValues)
                                {
                                    writer.WriteStartElement("Value");
                                    writer.WriteString(o.ToString());
                                    writer.WriteEndElement();
                                }

                                break;
                            }
                        case WiaSubType.RangeSubType:
                            {
                                writer.WriteAttributeString("Count", 3.ToString());
                                writer.WriteElementString("Min", prop.SubTypeMin.ToString());
                                writer.WriteElementString("Max", prop.SubTypeMax.ToString());
                                writer.WriteElementString("Step", prop.SubTypeStep.ToString());
                                break;
                            }

                        default:
                            {
                                break;
                            }

                    }
                    writer.WriteEndElement();
                }
                catch (Exception ex)
                {
                    writer.WriteString("Couldn't evaluate");
                }
                writer.WriteEndElement();
            }

            writer.Flush();
        }
    }
}

File: iCopySharp/Classes/PropertiesToXML.vb
Imports WIA
Imports System.Xml

Module PropertiesToXML

    Sub PropertiesToXML(properties As WIA.Properties, writer As XmlTextWriter)
        For Each prop As WIA.Property In properties

            writer.WriteStartElement("Property")
            Try

                writer.WriteAttributeString("ID", prop.PropertyID)
                writer.WriteAttributeString("Name", prop.Name)
                writer.WriteAttributeString("IsReadOnly", prop.IsReadOnly)
                writer.WriteAttributeString("Type", String.Format("{0}", DirectCast(prop.Type, WiaPropertyType)))
                writer.WriteAttributeString("Value", prop.Value.ToString())

                writer.WriteStartElement("Subtype")
                writer.WriteAttributeString("Type", String.Format("{0}", prop.SubType))
                Select Case prop.SubType
                    Case WiaSubType.UnspecifiedSubType
                        writer.WriteAttributeString("Count", 0)
                    Case WiaSubType.FlagSubType
                        writer.WriteAttributeString("Count", String.Format("{0}", prop.SubTypeValues.Count))
                        For Each o As Object In prop.SubTypeValues
                            writer.WriteStartElement("Flag")
                            writer.WriteString(o.ToString())
                            writer.WriteEndElement()
                        Next
                    Case WiaSubType.ListSubType
                        writer.WriteAttributeString("Count", String.Format("{0}", prop.SubTypeValues.Count))
                        For Each o As Object In prop.SubTypeValues
                            writer.WriteStartElement("Value")
                            writer.WriteString(o.ToString())
                            writer.WriteEndElement()
                        Next
                    Case WiaSubType.RangeSubType
                        writer.WriteAttributeString("Count", 3)
                        writer.WriteElementString("Min", prop.SubTypeMin.ToString())
                        writer.WriteElementString("Max", prop.SubTypeMax.ToString())
                        writer.WriteElementString("Step", prop.SubTypeStep.ToString())
                    Case Else

                End Select
                writer.WriteEndElement()
            Catch ex As Exception
                writer.WriteString("Couldn't evaluate")
            End Try
            writer.WriteEndElement()
        Next

        writer.Flush()
    End Sub
End Module
